components/button/stories/template.js

```diff
@@ -53,31 +57,39 @@ export const Template = ({
         [`${rootClass}--size${size?.toUpperCase()}`]: typeof size !== "undefined",
         [`${rootClass}--static${capitalize(lowerCase(staticColor))}`]: typeof staticColor !== "undefined",
         [`${rootClass}--iconOnly`]: hideLabel,
-        "is-pending": isPending,
+        ["is-pending"]: isPending,
+        ["is-disabled"]: isDisabled,
+				["is-hover"]: isHovered,
+				["is-focus-visible"]: isFocused,
+				["is-active"]: isActive,
         ...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
       })}
       id=${ifDefined(id)}
       style=${ifDefined(styleMap(customStyles))}
       ?disabled=${isDisabled}
-      @click=${!isPendingStory ? onclick : () => {
-        isPending
-          ? updateArgs({ isPending: false })
-          : updateArgs({ isPending: true });
+      @click=${onclick ?? function() {
+        // Toggle the is-pending state on-click
+        updateArgs({ isPending: true });
+        setTimeout(() => {
+          updateArgs({ isPending: false });
+        }, 1000);
```

Comment:
Could you bump up the time on this a little? After click, the switch back to default from pending happens a little fast. Perhaps around 3 to 5 seconds?